### PR TITLE
[CELEBORN-1239] Introduce default quota for DefaultQuotaManager

### PR DESCRIPTION
**Pull request by leixm**
### What changes were proposed in this pull request?
Introduce default quota for DefaultQuotaManager.

### Why are the changes needed?
Many times, for users who do not have a quota configured, we hope to have a default quota that applies to them.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
UTs.

### CODE REVIEW

**Comment by leixm** [comment]
cc @AngersZhuuuu

**Comment by leixm** [comment]
> What's the behavior of we didn't setting default quota?

As before, no quotas.

File: conf/quota.yaml.template [hunk 1]
```diff
@@ -30,3 +30,12 @@
...
      hdfsFileCount: 200
+
+# Users without configuration will use this configuration
+-  tenantId: default
+   name: default
+   quota:
+     diskBytesWritten: 1G
+     diskFileCount: 1000
+     hdfsBytesWritten: 1G
+     hdfsFileCount: 1000
```
Comment (AngersZhuuuu) [inline conf/quota.yaml.template:41]: Should not add to this template, should add this to quotas template

**Comment by AngersZhuuuu** [comment]
ping @SteNicholas @FMX @waitinfuture It's an important thing to do this before we can get actual user resource usage before set quota for each user.

File: conf/quota.yaml.template (hunk 1 above)
Comment (leixm) [reply conf/quota.yaml.template:41]: This is quota template.

Comment (AngersZhuuuu) [reply conf/quota.yaml.template:41]: Also need add this to doc

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks! cc @RexXiong PTAL

File: conf/quota.yaml.template (hunk 1 above)
Comment (leixm) [reply conf/quota.yaml.template:41]: okk.

**Review by AngersZhuuuu — APPROVED** [review]
LGTM

File: conf/quota.yaml.template (hunk 1 above)
Comment (leixm) [reply conf/quota.yaml.template:41]: done.

File: conf/quota.yaml.template [hunk 2]
```diff
@@ -30,3 +30,12 @@
      diskFileCount: 1000
      hdfsBytesWritten: 3G
      hdfsFileCount: 200
+
+# Users without configuration will use this configuration
```
Comment (RexXiong) [inline conf/quota.yaml.template:34]: IMO, we can add a new field `level` such as TENANT、USER（default for compatible）instead hack the name as default, and For cluster default, we can use SYSTEM level. 

>  -  level: System
   quota:
     diskBytesWritten: 100m
     diskFileCount: 200
     hdfsBytesWritten: 200m
     hdfsFileCount: 100

Comment (leixm) [reply conf/quota.yaml.template:34]: Good idea.

Comment (leixm) [reply conf/quota.yaml.template]: cc @RexXiong   PTAL.

Comment (leixm) [reply conf/quota.yaml.template]: I modified the hierarchy of the configuration file to make it more intuitive.

**Pull request closed without merging**